proxy/src/control/mod.rs

```diff
@@ -68,34 +66,36 @@ impl Background {
         self,
         host_and_port: HostAndPort,
         dns_config: dns::Config,
-        executor: &Handle,
-    ) -> Box<Future<Item = (), Error = ()>> {
+    ) -> Box<Future<Item = (), Error = ()>>
+    {
         // Build up the Controller Client Stack
         let mut client = {
             let ctx = ("controller-client", format!("{:?}", host_and_port));
             let scheme = http::uri::Scheme::from_shared(Bytes::from_static(b"http")).unwrap();
             let authority = http::uri::Authority::from(&host_and_port);
-            let dns_resolver = dns::Resolver::new(dns_config, &executor);
+            let dns_resolver = dns::Resolver::new(dns_config);
             let connect = Timeout::new(
-                LookupAddressAndConnect::new(host_and_port, dns_resolver, executor),
+                LookupAddressAndConnect::new(host_and_port, dns_resolver),
                 Duration::from_secs(3),
-                &executor,
-            );
+            ).named("looking up controller address and connecting");
 
             let h2_client = tower_h2::client::Connect::new(
                 connect,
                 h2::client::Builder::default(),
-                ::logging::context_executor(ctx, executor.clone()),
+                ::logging::context_executor(
+                    ctx,
+                    current_thread::TaskExecutor::current()
```

Comment:
This is the first time I see `current_thread` referenced in this PR. Could you explain why you opted for that here?

Reply:
In most cases, it's because I didn't want to put `Send` bounds on everything or had some type that couldn't be `Send`. I think it's not actually necessary in this particular instance, though (I think it was at one point); this can probably be changed to `DefaultExecutor`.

I thought that it was probably correct to use `current_thread` explicitly in the control thread code, since I doubt we will ever want to run the control work on a threadpool?